common/scala/src/main/scala/org/apache/openwhisk/core/entity/ActivationResult.scala

```diff
@@ -45,6 +48,15 @@ protected[core] case class ActivationResponse private (val statusCode: Int, val
   def isSuccess = statusCode == ActivationResponse.Success
   def isApplicationError = statusCode == ActivationResponse.ApplicationError
   def isContainerError = statusCode == ActivationResponse.DeveloperError
+  def isContainerTimeoutError(timeout: FiniteDuration) = {
+    statusCode == ActivationResponse.DeveloperError &&
+    (result.get match {
+      case JsObject(fields) =>
+        fields.contains(ActivationResponse.ERROR_FIELD) && fields(ActivationResponse.ERROR_FIELD)
+          .toString() == '"' + Messages.timedoutActivation(timeout, false) + '"'
```

Comment:
Please never use `toString` on JSON. It get's very unwieldy and bypasses typechecks (since your just breaking everything down to a string).

As these are all case-classes, you should be able to greatly simplify the whole check to

```scala
def isContainerTimeoutError(timeout: FiniteDuration) = this == ActivationResponse.developerError(Messages.timedoutActivation(timeout, false))
```

Does that work?

Reply:
I'd even say: take that piece of code to the caller side if it can be simplified like proposed.

Reply:
done, handled the case in the new method `isTimedoutActivation` in `WhiskActivation`